projects/igniteui-angular/src/lib/date-range-picker/igx-date-range-picker.component.ts

```diff
@@ -0,0 +1,696 @@
+     * ```html
+     * <igx-date-range-picker (onOpening)="handleOpening($event)"></igx-date-range-picker>
+     * ```
+     */
+    @Output()
+    public onOpening = new EventEmitter<CancelableBrowserEventArgs & IBaseEventArgs>();
+
+    /**
+     * Emitted when the `IgxDateRangeComponent` is opened.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker (onOpened)="handleOpened($event)"></igx-date-range-picker>
+     * ```
+     */
+    @Output()
+    public onOpened = new EventEmitter<IBaseEventArgs>();
+
+    /**
+     * Emitted when the calendar starts closing, cancelable.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker (onClosing)="handleClosing($event)"></igx-date-range-picker>
+     * ```
+     */
+    @Output()
+    public onClosing = new EventEmitter<CancelableBrowserEventArgs & IBaseEventArgs>();
+
+    /**
+     * Emitted when the `IgxDateRangeComponent` is closed.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker (onClosed)="handleClosed($event)"></igx-date-range-picker>
+     * ```
+     */
+    @Output()
+    public onClosed = new EventEmitter<IBaseEventArgs>();
+
+    /** @hidden */
+    @HostBinding('class.igx-date-range-picker')
+    public cssClass = 'igx-date-range-picker';
+
+    /** @hidden */
+    @ViewChild(IgxDateRangeSingleComponent)
+    public single: IgxDateRangeSingleComponent;
+
+    /** @hidden */
+    @ViewChild(IgxCalendarComponent)
+    public calendar: IgxCalendarComponent;
+
+    /** @hidden */
+    @ViewChild(IgxToggleDirective)
+    public toggleDirective: IgxToggleDirective;
+
+    /** @hidden */
+    @ContentChildren(IgxPickerToggleComponent, { descendants: true })
+    public toggleComponents: QueryList<IgxPickerToggleComponent>;
+
+    /** @hidden */
+    @ContentChildren(IgxInputGroupBase)
+    public projectedInputs: QueryList<IgxInputGroupBase>;
+
+    @ContentChild(IgxLabelDirective)
+    public label: IgxLabelDirective;
+
+    /** @hidden */
+    @ContentChild(IgxDateRangeSeparatorDirective, { read: TemplateRef })
+    public dateSeparatorTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    public dateSeparator = CurrentResourceStrings.DateRangePickerResStrings.igx_date_range_picker_date_separator;
+
+    /** @hidden @internal */
+    public get appliedFormat() {
+        // TODO: apply formatter if present
+        if (!this.hasProjectedInputs) {
+            // TODO: use displayFormat - see how shortDate, longDate can be defined
+            return this.inputFormat
+                ? `${this.inputFormat} - ${this.inputFormat}`
+                : `${DEFAULT_INPUT_FORMAT} - ${DEFAULT_INPUT_FORMAT}`;
+        } else {
+            return this.inputFormat ? this.inputFormat : DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public get hasProjectedInputs(): boolean {
+        return this.projectedInputs.some(i => i instanceof IgxDateRangeStartComponent || i instanceof IgxDateRangeEndComponent);
```

Comment:
Is this really needed? The `ContentChildren` query list should only contain `IgxDateRangeStart/EndComponents`, since the template does not have a "catch all" `ng-content` outlet